src/orchard/apropos.clj

```diff
@@ -15,43 +14,41 @@
 
 ;;; ## Symbol Search
 
-(defn namespaces
-  "Return the list of namespaces to be searched, ordered with `ns` first,
-  followed by `clojure.*` namespaces, and then all others sorted alphabetically.
-  If `search-ns` is specified, the returned list will contain only this
-  namespace. `filter-regexps` is used to filter out namespaces matching regexps."
-  [ns search-ns & [filter-regexps]]
-  (let [clojure-ns? #(.startsWith (str (ns-name %)) "clojure.")
-        current-ns (find-ns (symbol (or ns "")))]
-    (if search-ns
-      (list (find-ns (symbol search-ns)))
-      (->> (all-ns)
-           (sort (fn [x y]
-                   (cond (= x current-ns) -1
-                         (= y current-ns)  1
-                         (and (clojure-ns? x) (not (clojure-ns? y))) -1
-                         (and (clojure-ns? y) (not (clojure-ns? x)))  1
-                         :else (compare (str x) (str y)))))
-           (remove ns/inlined-dependency?)
-           (remove #(ns/internal-namespace? % filter-regexps))))))
+(defn- apropos-sort
+  "Return a list of vars, ordered with `ns` first,
+  followed by `clojure.*` namespaces, and then all others sorted
+  alphabetically."
+  [ns vars]
+  (let [clojure-ns? #(.startsWith (str (ns-name %)) "clojure.")]
+    (sort-by
+      (comp :ns meta)
+      (fn [x y]
+        (cond
+          (nil? x) 1
+          (nil? y) -1
+          (= x ns) -1
+          (= y ns)  1
+          (and (clojure-ns? x) (not (clojure-ns? y))) -1
+          (and (clojure-ns? y) (not (clojure-ns? x)))  1
+          :else (compare (str x) (str y))))
+      vars)))
 
 (defn find-symbols
   "Find symbols or (optionally) docstrings matching `query` in `search-ns` if
   specified or all namespaces. The search may optionally include private vars,
   and may be case senstive. Types returned correspond to Apropos types.
   Docstring search returns the full doc; symbol search returns an abbreviated
   version."
-  [{:keys [ns query search-ns docs? privates? case-sensitive? filter-regexps]}]
```

Comment:
You've changed the signature of this function, but you didn't change its docstring.